tools/REALTEK_RTL8195AM.py

```diff
@@ -0,0 +1,161 @@
+"""
+mbed REALTEK_RTL8195AM elf2bin script
+Copyright (c) 2011-2016 Realtek Semiconductor Corp.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+LIBRARIES BUILD
+"""
+
+import sys, array, struct, os, re, subprocess
+import hashlib
+
+from tools.paths import TOOLS_BOOTLOADERS
+from datetime import datetime
+
+# Constant Variables
+RAM2_RSVD = 0x3131373835393138
+
+def write_fixed_width_string (value, width, output):
+        # cut string to list & reverse
+        line = [value[i:i+2] for i in range(0, len(value), 2)]
+        output.write("".join([chr(long(b, 16)) for b in line]))
+
+def write_fixed_width_value (value, width, output):
+        # convert to string
+        line = format(value, '0%dx' %(width))
+        if len(line) > width:
+            print "[ERROR] value 0x%s cannot fit width %d" %(line, width)
+            sys.exit(-1)
+        # cut string to list & reverse
+        line = [line[i:i+2] for i in range(0, len(line), 2)]
+        line.reverse()
+        # convert to write buffer
+        output.write("".join([chr(long(b, 16)) for b in line]))
+
+def append_image_file (image, output):
+    input = open (image, "rb")
+    output.write(input.read())
+    input.close()
+
+# ----------------------------
+#       main function
+# ----------------------------
+def prepend (image, sym_addr, image_prepend):
+
+    # parse input arguments
+    offset = 44
+
+    output_file = image_prepend
+    output = open (output_file, "wb")
+
+    write_fixed_width_value (os.stat(image).st_size, 8, output)
+    write_fixed_width_value (int(sym_addr), 8, output)
+    write_fixed_width_value (RAM2_RSVD, 16, output)
+
+    append_image_file (image, output)
+    output.close()
+
+def find_section(toolchain, elf, section):
```

Comment:
It looks like this function is a bit of copy paste code I would re-write it as follows:

```python


def find_section(toolchain, elf, section):
    mapfile = os.path.splitext(".")[0] + ".map"
    if toolchain == "GCC_ARM":
        section_re = re.compile(r'^' + section + r'\s+0x0{,8}(?P<addr>[0-9A-Fa-f]{8})\s+.*$')
    elif toolchain in ["ARM_STD", "ARM", "ARM_MICRO"]:
        section_re = re.compile(r'^.*Region\s+' + section + r'\s+\(Base: 0x(?P<addr>[0-9A-Fa-f]{8}),\s+Size:.*\)$')
    elif toolchain == "IAR":
        section_re = re.compile(r'^\s+' + section + r'\s+0x(?P<addr>[0-9A-Fa-f]{8})\s+.*<Block>$')
    else:
        print "[ERROR] unsupported toolchain " + toolchain
        sys.exit(-1)
    for line in open(mapfile):
        match = re.match(section_re, line)
        if match:
            return int(mach.group("addr"), 16)
    print "[ERROR] cannot find the address of section " + section
    return 0
```